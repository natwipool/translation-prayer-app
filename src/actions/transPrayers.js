// SET_TRANS_PRAYER
const setTransPrayer = (transPrayers = {}) => ({
  type: 'SET_TRANS_PRAYER',
  transPrayers
});

export default setTransPrayer;