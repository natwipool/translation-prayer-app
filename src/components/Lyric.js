import React from 'react';
import EachLyric from './EachLyric';

const Lyric = (props) => {
  return (
    <div>
      <h3>{props.transPrayer.precept}</h3>
      {props.transPrayer.lyrics.map((lyric, index) => 
        <EachLyric 
          key={index}
          lyric={lyric}
        />
      )}
    </div>
  )
}

export default Lyric;