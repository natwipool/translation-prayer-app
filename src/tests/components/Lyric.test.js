import React from 'react';
import { shallow } from 'enzyme';
import Lyric from '../../components/Lyric';
import transPrayers from '../fixtures/transPrayers';

test('should render Lyric correctly', () => {
  const wrapper = shallow(<Lyric transPrayer={transPrayers[2]}/>);
  expect(wrapper).toMatchSnapshot();
});