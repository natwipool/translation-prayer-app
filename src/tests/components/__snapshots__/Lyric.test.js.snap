// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Lyric correctly 1`] = `
<div>
  <h3>
    สัพพปัตติทานคาถา
  </h3>
  <EachLyric
    key="0"
    lyric="(นำ) หันทะ มะยัง สัพพะปัตติทานะคาถาโย ภะณามะ เส."
  />
  <EachLyric
    key="1"
    lyric="ปุญญัสสิทานิ กะตัสสะ ยานัญญานิ กะตานิ เม, เตสัญจะ ภาคิโน โหนตุ สัตตานันตาปปะมาณะกา"
  />
  <EachLyric
    key="2"
    lyric="สัตว์ทั้งหลายไม่มีที่สุดไม่มีประมาณ, จงมีส่วนแห่งบุญที่ข้าพเจ้าได้ทำในบัดนี้, และแห่งบุญอื่นที่ได้ทำไว้ก่อนแล้ว"
  />
  <EachLyric
    key="3"
    lyric="เย ปิยา คุณะวันตา จะ มัยหัง มาตาปิตาทะโย, ทิฏฐา เม จาปะยาทิฏฐา วา อัญเญ มัชฌัตตะเวริโน"
  />
  <EachLyric
    key="4"
    lyric="คือจะเป็นสัตว์เหล่าใด, ซึ่งเป็นที่รักใคร่และมีบุญคุณ เช่นมารดา บิดาของข้าพเจ้าเป็นต้นก็ดี, ที่ข้าพเจ้าเห็นแล้วหรือไม่ได้เห็นก็ดี, สัตว์เหล่าอื่นที่เป็นกลางๆหรือเป็นคู่เวรกันก็ดี"
  />
  <EachLyric
    key="5"
    lyric="สัตตา ติฏฐันติ โลกัสมิง เตภุมมา จะตุโยนิกา, ปัญเจกะจะตุโวการา สังสะรันตา ภะวาภะเว"
  />
  <EachLyric
    key="6"
    lyric="สัตว์ทั้งหลายตั้งอยู่ในโลก, อยู่ในภูมิทั้งสาม, อยู่ในกำเนิดทั้งสี่, มีขันธ์ห้าขันธ์ มีขันธ์ขันธ์เดียว มีขันธ์สี่ขันธ์, กำลังท่องเที่ยวอยู่ในภพน้อยภพใหญ่ก็ดี"
  />
  <EachLyric
    key="7"
    lyric="ญาตัง เย ปัตติทานัมเม อะนุโมทันตุ เต สะยัง, เย จิมัง นัปปะชานันติ เทวา เตสัง นิเวทะยุง"
  />
  <EachLyric
    key="8"
    lyric="สัตว์เหล่าใดรู้ส่วนบุญที่ข้าพเจ้าแผ่ให้แล้ว, สัตว์เหล่านั้นจงอนุโมทนาเองเถิด, ส่วนสัตว์เหล่าใดยังไม่รู้ส่วนบุญนี้, ขอเทวดาทั้งหลาย จงบอกสัตว์เหล่านั้นให้รู้"
  />
  <EachLyric
    key="9"
    lyric="มะยา ทินนานะ ปุญญานัง อะนุโมทะนะเหตุนา, สัพเพ สัตตา สะทา โหนตุ อะเวรา สุขะชีวิโน, เขมัปปะทัญจะ ปัปโปนตุ เตสาสา สิชฌะตัง สุภา"
  />
  <EachLyric
    key="10"
    lyric="เพราะเหตุที่ได้อนุโมทนาส่วนบุญที่ข้าพเจ้าแผ่ให้แล้ว, สัตว์ทั้งหลายทั้งปวง, จงเป็นผู้ไม่มีเวรอยู่เป็นสุขทุกเมื่อ, จงถึงบทอันเกษมกล่าวคือพระนิพพาน, ความปรารถนาที่ดีงามของสัตว์เหล่านั้นจงสำเร็จเถิด"
  />
</div>
`;
